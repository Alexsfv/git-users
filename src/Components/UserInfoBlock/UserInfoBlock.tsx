import React from 'react'


type UserInfoBlockProps = {
    title: string
    data: string | null | number
}
const UserInfoBlock: React.FC<UserInfoBlockProps> = ({data, title}) => {
    const stringData = data + ''
    const isHidden = stringData === 'false' || stringData === 'null' || stringData === 'undefined' || stringData === ''

    if (!isHidden) {
        return (
            <div className="d-flex w-100 p-3 mb-3 bg-light rounded shadow text-dark font-weight-bold">
                <p className="w-100 m-0" style={{fontSize: '18px'}}>{title}: </p>
                <p className="w-100 m-0 text-right">
                    {
                        stringData.startsWith('http')
                            ? <a href={stringData}>{stringData}</a>
                            : stringData
                    }
                </p>
            </div>
        )
    }
    return null
}

export default UserInfoBlock