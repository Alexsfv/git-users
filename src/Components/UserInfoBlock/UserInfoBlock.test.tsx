import { render, shallow } from 'enzyme'
import React from 'react'
import UserInfoBlock from './UserInfoBlock'

describe("UserInfoBlock component", () => {
    it('should render UserInfoBlock', () => {
        const component = shallow(<UserInfoBlock title="ssd" data="11-12-11"/>)
        expect(component).toMatchSnapshot()
    })

    it("should render UserInfoBlock with link", () => {
        const component = shallow(<UserInfoBlock title="ssd" data="https://google.com"/>)
        expect(component).toMatchSnapshot()
    })

    it("should empty render UserInfoBlock", () => {
        const component = shallow(<UserInfoBlock title="title" data={null}/>)
        expect(component.isEmptyRender()).toBe(true)
    })
})