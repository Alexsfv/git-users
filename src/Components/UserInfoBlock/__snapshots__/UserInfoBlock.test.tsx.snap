// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfoBlock component should render UserInfoBlock 1`] = `
<div
  className="d-flex w-100 p-3 mb-3 bg-light rounded shadow text-dark font-weight-bold"
>
  <p
    className="w-100 m-0"
    style={
      Object {
        "fontSize": "18px",
      }
    }
  >
    ssd
    : 
  </p>
  <p
    className="w-100 m-0 text-right"
  >
    11-12-11
  </p>
</div>
`;

exports[`UserInfoBlock component should render UserInfoBlock with link 1`] = `
<div
  className="d-flex w-100 p-3 mb-3 bg-light rounded shadow text-dark font-weight-bold"
>
  <p
    className="w-100 m-0"
    style={
      Object {
        "fontSize": "18px",
      }
    }
  >
    ssd
    : 
  </p>
  <p
    className="w-100 m-0 text-right"
  >
    <a
      href="https://google.com"
    >
      https://google.com
    </a>
  </p>
</div>
`;
